Tue May 13 22:07:19 FET 2014  Sergei Trofimovich <slyfox@community.haskell.org>
  * resolve issue2364: don't break list of 'bad sources'
  
  This time the bug manifested on a simple operation:
  $ darcs record -a -m "something"
  
  Attempt to write a patch resulted in something like:
    Failed to record patch 'hello'
  
    HINT: I could not reach the following repositories:
          http://repetae.net/repos/jhc
          /home/st/.darcs/cache
          /home/st/.cache/darcs
          /home/st/dev/darcs/jhc
        If you're not using them, you should probably delete
  
  The sequence should be the following:
      1. store patch to inventory/foo
      2. try to store to a writable cache (say, ~/.darcs/cache/patches)
      3. fail to write
      4. filter out bad caches
      5. try again
      6. copy from cache to patches/
  
  Due to missing NOINLINE step 4. led to
  all caches treated as writable, thus step 5
  failed without a chance for patch to
  go to 'patches/'.
  
  As a side-effect building darcs with -O0 produced seemingly working darcs.
  Reported-by: Ivan Miljenovic
diff -rN -u old-darcs.net/src/Darcs/Util/GlTue May 13 22:07:19 FET 2014  Sergei Trofimovich <slyfox@community.haskell.org>
  * resolve issue2364: don't break list of 'bad sources'
  
  This time the bug manifested on a simple operation:
  $ darcs record -a -m "something"
  
  Attempt to write a patch resulted in something like:
    Failed to record patch 'hello'
  
    HINT: I could not reach the following repositories:
          http://repetae.net/repos/jhc
          /home/st/.darcs/cache
          /home/st/.cache/darcs
          /home/st/dev/darcs/jhc
        If you're not using them, you should probably delete
  
  The sequence should be the following:
      1. store patch t